lib/OpenSearch/Parameters/Document/Index.pm

```diff
@@ -1,174 +1,89 @@
 package OpenSearch::Parameters::Document::Index;
-use Moose::Role;
-use Moose::Util::TypeConstraints;
-enum 'DocumentOpType'      => [qw(index create)];
-enum 'DocumentRefresh'     => [qw(true false wait_for)];
-enum 'DocumentVersionType' => [qw(internal external external_gte)];
+use strict;
+use warnings;
+use feature qw(state);
+use Types::Standard qw(Str Bool Int Enum HashRef);
+use Types::Common::String qw(NonEmptyStr);
+use Moo::Role;
+
+with 'OpenSearch::Parameters';
 
 has 'index' => (
   is          => 'rw',
-  isa         => 'Str',
-  metaclass   => 'MooseX::MetaDescription::Meta::Attribute',
-  description => {
-    encode_func => 'as_is',
-    type        => 'path',
-    required    => 1,
-  }
+  isa         => NonEmptyStr,
+  required    => 1,
 );
 
 # create if _doc does not exist
```

Comment:
This is merely a flag to switch between update and insert. One reason not to just generate the Moo-parameter-list from one single specification. Now it is not processed because all `path` are processed by the executor. Not?

Reply:
Yes. Currently all path parameters are handled in the `execute` method. Its a "made up" parameter to allow for the specification in the Opensearch Docs:
`_create is a type identifier indicating that document creation should only occur if the document with the specified ID doesn’t already exist.`

Path parameters have never been used outside of the execute method. They were only there to handle required path parameters. I never used the "required" flag of attributes in Moo(se). Required parameters were handled in `OpenSearch::Helper` to provide an "own" error message on missing params